### ISSUE TITLE
xcom2 can't work with current version legendary alone

### ISSUE THREAD

**Issue by Trokinos** [derrod/legendary#422]
Using Window 10 and  latest "0.20.25 - Our Benefactors" version of Legendary.

This week on the epic store they are giving away XCOM 2

After downloading it with Legendary , i am then launching it with
C:\Legendary\legendary.exe launch thegamecode
(with "thegamecode" being of course the game i see in the Legendary list of game i own on my epic store account)
This had always worked for all the game i obtained on the epic store 

But xcom 2 is completely dependant for its start to a "2K Launcher", it's simply not possible to directly launch the game.
So launching the game through Legendary will then start as expected  the  2K launcher.
But from there  there is no way to start the acctual game because the 2K Launcher does not find the epic launcher  (of course as i only use Legendary, i don't have the epic launcher on my system) and only offer you to download it.

For the steam version, there exist an "alternative launcher" apparently to bypass the 2K one, but still thislauncher require the steam application to be loaded, meaning that even if they added epic store support there would still be a need of the epic launcher installed.

**Comment by braiam** [comment]
Have you tried the `--override-exe Binaries/Win64/XCom2.exe` option?

**Comment by derrod** [comment]
Technically the same as #402, I don't know what the Epic Launcher does (or Legendary doesn't) for the 2K launcher to recognise it as installed. It's kinda weird.

**Comment by Zobrob** [comment]
> Technically the same as #402, I don't know what the Epic Launcher does (or Legendary doesn't) for the 2K launcher to recognise it as installed. It's kinda weird.

I just wanted to comment that civ6 is not working for me. You can use the direct .exe without the 2k launcher but this has several problems. The game starts but no multiplayer and no "New Frontier" DLC is available. However, if I try to use the 2K launcher it always says in the launcher that the game is not installed. I already tried reinstalling the game, running as admin and reinstalling the 2K launcher (none worked).

**Comment by Trokinos** [comment]
> Have you tried the `--override-exe Binaries/Win64/XCom2.exe` option?

It works !
I launched the game with 
`C:\Legendary\legendary.exe launch thegamecode --override-exe Binaries/Win64/XCom2.exe`
Launching the game this way allowed me to completely bypass that useless 2K Launcher and directly launch the game itself.

**Comment by braiam** [comment]
> > Technically the same as #402, I don't know what the Epic Launcher does (or Legendary doesn't) for the 2K launcher to recognise it as installed. It's kinda weird.
> 
> I just wanted to comment that civ6 is not working for me. You can use the direct .exe without the 2k launcher but this has several problems. The game starts but no multiplayer and no "New Frontier" DLC is available (treated as Epic offline mode). However, if I try to use the 2K launcher it always says in the launcher that the game is not installed. I already tried reinstalling the game, running as admin and reinstalling the 2K launcher (none worked).

Same as Trokino, have you tried --override-exe? That option still uses legendary and adds the parameters that are normally used for the launcher for the game itself.

**Comment by Zobrob** [comment]
> 

Yes, I tried the solution of [402](https://github.com/derrod/legendary/issues/402) before. With everything I tried I either got Civ6 running in offline mode, with just singleplayer enabled (which is probably enough for OP) and with the last DLC missing, or I got a 2K launcher window, that told me I own everything but the content is not currently installed. A click on 'install' then wants to open a link with the Epic launcher.
I guess my issue is not as related to OPs question as I first thought, so this can be closed or moved. Maybe in future another user can either confirm that only offline mode is working or that I have a misconfiguration on my side. Thanks

**Comment by Trokinos** [comment]
I added a page on the wiki for how to launch XCOM2 with Legendary by using the
`--override-exe Binaries/Win64/XCom2.exe`  
But as it's my first time doing so and i'm not english speaker, i would appreciate if someone could check if i did it correctly and if it is has not horrible grammar.

**Comment by derrod** [comment]
> Yes, I tried the solution of [402](https://github.com/derrod/legendary/issues/402) before. With everything I tried I either got Civ6 running in offline mode, with just singleplayer enabled (which is probably enough for OP), or I got a 2K launcher window, that told me I own everything but the content is not currently installed. A click on 'install' then wants to open a link with the Epic launcher. I guess my issue is not as related to OPs question as I first thought, so this can be closed or moved. Maybe in future another user can either confirm that only offline mode is working or that I have a misconfiguration on my side. Thanks

I tried CIV 6 with an exe override to `Base/Binaries/Win64EOS/CivilizationVI.exe` and that worked, including multiplayer (on Windows at least).

**Comment by derrod** [comment]
So as an update, this is a bit more complicated than I thought.

When the game is launched the following happens:
1. EGL runs 2K Launcher
2. 2K Launcher uses URL to requrest EGL to run the game
3. 2K Launcher waits to be called again
4. Game finally gets launched

I still don't know what actually leads to the 2K launcher recognising the game as installed, but even if that works you couldn't quite launch the game as it would try to run the Epic Games Launcher regardless. So just bypassing it is your best bet for now. Although I don't know why online authentication for CIV does not appear to work on Linux.

**Comment by derrod** [comment]
Starting with the next version, Legendary will warn users about the 2K launcher, and automatically override the EXE for known games: https://github.com/derrod/legendary/commit/a3bc07e15a13a67931a9d10c84dc978a55b15f2b

I'll also add a warning on launch about the launcher, perhaps including a suggestions for alternative executables.
